FortranCon2021-stdlib/stdlib-talk.tex

```diff
@@ -0,0 +1,302 @@
+call global_logger%log_information('Something informative')
+call global_logger%log_error('Oopsie daisy')
+end
+\end{verbatim}
+  \end{block}
+  \begin{block}{log.txt}
+    \small
+\begin{verbatim}
+2021-09-13 23:31:30.346: INFO: Something informative
+2021-09-13 23:31:30.346: ERROR: Oopsie daisy
+\end{verbatim}
+  \end{block}
+
+  \note{
+    This first example is a quick demonstration of the new `stdlib_logger` module.
+    This module defines a derived type for logging messages to a file.
+    This logger type is highly configurable, with options to control which file (or files) the logger should write to, how to format time stamps, which types of messages to log, and lots more.
+    For very basic needs, one can use the predefined `global_logger` instance, as done here.
+    Here, we're just connecting the logger to a file on disk and writing messages to it.
+    Note how the ``debug'' message was not written, since the global logger is configured by default to log ``info'', ``warnings'', and ``errors''.
+    This is just scratching the surface; the design is flexible enough to set up fairly sophisticated logging infrastructure.
+    For instance, in a coarray application, one might configure separate loggers for each image or team of images.
+    stdlib now makes it easy to set that up.
+  }
+\end{frame}
+
+
+\begin{frame}[fragile]
+  \frametitle{Demo: stdlib\_bitsets}
+
+  \begin{block}{ex\_bitsets.f90}
+    \small
+\begin{verbatim}
+use stdlib_bitsets
+implicit none
+
+integer :: i; type(bitset_64) :: b1, b2
+
+call b1%from_string('001100') ! S6B001110
+b2 = [(.true., i=1,6)]        ! S6B111111
+
+call xor(b1, b2)              ! S6B110001, S6B111111
+
+call b1%set(2, 4)             ! S6B111111 -- N.B. 0-based index
+print *, b1 == b2             ! T
+end
+\end{verbatim}
+  \end{block}
+
+  \note{
+    The second demo is of the bitsets module.
+    This provides a feature that is often proposed for standardization: an abstract type for working with strings of bits.
+    The basic bitset type uses a 64-bit integer internally, but it keeps track of the actual length of the bitset being represented.
+    Here, I'm initializing two 6-bit sets, one from a string and one from an array of logicals.
+    The comment shows a string representation of the bitset, which can be taken as a bitset 'literal' for the purposes of I/O.
+    Here, the prefix 'S6B' means a set of 6 bits.
+    Most operations with bitsets work in-place rather than creating new instances.
+    For instance, all the usual boolean algebra operations -- like the exclusive-or here -- are implemented as subroutines that overwrite the first argument.
+    Likewise, when one sets or clears single bits or contiguous ranges of bits, the data is modified without making a copy.
+  }
+\end{frame}
+
+
+\begin{frame}
+  \frametitle{New contributors have been key to stdlib's growth}
+
+  \begin{itemize}
+  \item From 16 committors to 34, including our 2 GSoC students
```

Comment:
Better use https://fortran-lang.org/community/ to count contributors, since it includes those participating in the discussions at the issue tracker and the code review. The actual code contributors can be a separate point.

Reply:
Thanks! I was trying to find this on GitHub, forgetting that it's on our own website!